.github/workflows/build.yaml

```diff
@@ -0,0 +1,32 @@
+name: Build
+
+on:
+  pull_request:
+  push:
+
+defaults:
+  run:
+    shell: 'bash -Eeuo pipefail -x {0}'
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - name: Build
+        uses: snapcore/action-build@v1
+        id: snapcraft
+        with:
+          snapcraft-args: snap --output docker_${{ github.run_id}}.snap
+
+      - name: Upload
+        # TODO: the goal of this "if:" condition is to avoid creating artifacts on private forks (and using up all their storage quota),
+        # but Tianon couldn't find a clean way to do that so this was the next best thing
+        if: github.repository == 'canonical/docker-snap'
```

Comment:
```suggestion
```
If the artifact isn't uploaded on forks, the smoke test would fail when trying to download it.

Reply:
Yeah, that's true, done!